src/parser_v3.py

```diff
@@ -0,0 +1,616 @@
+#!/usr/bin/env python3
+
+import csv
+import json
+import requests
+from collections import defaultdict, OrderedDict
+from datetime import datetime, timedelta
+from pathlib import Path
+
+RESOURCE_DIR = Path('lib')
+# Contains list of districts
+DISTRICT_LIST = RESOURCE_DIR / 'district_list.csv'
+# Contains district data on 26th April
+OLD_DISTRICT_DATA = RESOURCE_DIR / 'districts_26apr_gospel.csv'
+DATE_END_OLD = '2020-04-26'
+
+INPUT_DIR = Path('tmp')
+# All raw_data's
+RAW_DATA = 'raw_data{n}.json'
+# Contains deaths and recoveries for entries in raw_data1 and raw_data2
+OUTCOME_DATA = 'deaths_recoveries{n}.json'
+# India testing data
+ICMR_TEST_DATA = INPUT_DIR / 'data.json'
+# States testing data
+STATE_TEST_DATA = INPUT_DIR / 'state_test_data.json'
+## For adding metadata
+# For state notes and last updated
+STATE_WISE = INPUT_DIR / 'data.json'
+# For district notes
+DISTRICT_WISE = INPUT_DIR / 'state_district_wise.json'
+
+OUTPUT_DIR = Path('tmp', 'v3')
+OUTPUT_MIN_DIR = OUTPUT_DIR / 'min'
+OUTPUT_DATA_FILENAME = 'data'
+OUTPUT_TIMESERIES_FILENAME = 'timeseries'
+
+STATE_CODES = {
+    'Andhra Pradesh': 'AP',
+    'Arunachal Pradesh': 'AR',
+    'Assam': 'AS',
+    'Bihar': 'BR',
+    'Chhattisgarh': 'CT',
+    'Goa': 'GA',
+    'Gujarat': 'GJ',
+    'Haryana': 'HR',
+    'Himachal Pradesh': 'HP',
+    'Jharkhand': 'JH',
+    'Karnataka': 'KA',
+    'Kerala': 'KL',
+    'Madhya Pradesh': 'MP',
+    'Maharashtra': 'MH',
+    'Manipur': 'MN',
+    'Meghalaya': 'ML',
+    'Mizoram': 'MZ',
+    'Nagaland': 'NL',
+    'Odisha': 'OR',
+    'Punjab': 'PB',
+    'Rajasthan': 'RJ',
+    'Sikkim': 'SK',
+    'State Unassigned': 'UN',
+    'Tamil Nadu': 'TN',
+    'Telangana': 'TG',
+    'Tripura': 'TR',
+    'Uttarakhand': 'UT',
+    'Uttar Pradesh': 'UP',
+    'West Bengal': 'WB',
+    'Andaman and Nicobar Islands': 'AN',
+    'Chandigarh': 'CH',
+    'Dadra and Nagar Haveli and Daman and Diu': 'DN',
+    'Delhi': 'DL',
+    'Jammu and Kashmir': 'JK',
+    'Ladakh': 'LA',
+    'Lakshadweep': 'LD',
+    'Puducherry': 'PY',
+    'Total': 'TT',
+    # To accomodate for improper entries in old data
+    #  'Dadra and Nagar Haveli': 'DN',
+    #  'Daman And Diu': 'DN',
+}
+
+DISTRICTS_DICT = defaultdict(dict)
+DISTRICTS_ADDITIONAL = {
+    'bsf camp': 'BSF Camp',
+    'italians': 'Italians',
+    'other state': 'Other State',
+    'railway quarantine': 'Railway Quarantine',
+    'airport quarantine': 'Airport Quarantine',
+    'evacuees': 'Evacuees',
+    'foreign evacuees': 'Foreign Evacuees',
+    'unassigned': 'Unassigned',
+}
```

Comment:
Can this be automated instead of having this in the file as a const? Modifying the code when a new category comes up is not desirable.

Reply:
So these are here just to filter down the logs. Unrecognized district fields won't cause the script to crash. Just run the script once. You'll understand what I'm saying. 

Coming back to the point, yes, I guess it can be either automated if it's published from the sheet, or it can be kept in a separate json/csv file in lib/.